src/games/strategy/engine/framework/ui/NewGameChooserEntry.java

```diff
@@ -51,31 +59,34 @@ public NewGameChooserEntry(final URI uri)
   }
 
   public void fullyParseGameData() throws GameParseException {
+    // TODO: We should be setting this in the the constructor. At this point, you have to call methods in the
+    // correct order for things to work, and that is bads.
     m_data = null;
 
-    String error = null;
     final AtomicReference<String> gameName = new AtomicReference<>();
 
-    try (InputStream input = m_url.toURL().openStream()) {
+    Optional<InputStream> inputStream = UrlStreams.openStream(m_url);
+    if(!inputStream.isPresent()) {
+      return;
+    }
+
+    try (InputStream input = inputStream.get()) {
       m_data = new GameParser().parse(input, gameName, false);
       m_gameDataFullyLoaded = true;
 
     } catch (final EngineVersionException e) {
       ClientLogger.logQuietly(e);
-      error = e.getMessage();
+      throw new GameParseException(e.getMessage());
     } catch (final SAXParseException e) {
       String msg = "Could not parse:" + m_url + " error at line:" + e.getLineNumber() + " column:" + e.getColumnNumber();
       ClientLogger.logError(msg);
       ClientLogger.logQuietly(e);
-      error = e.getMessage();
+      throw new GameParseException(e.getMessage());
     } catch (final Exception e) {
       String msg = "Could not parse:" + m_url;
       ClientLogger.logError(msg);
       ClientLogger.logQuietly(e);
```

Comment:
What about merging Line 86, 87 and 88 to? `ClientLogger.logError("Could not parse:" + m_url, e);` or `ClientLogger.logQuietly("Could not parse:" + m_url, e);`
Same a couple lines above